src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -219,7 +219,7 @@ private static bool TryGetByteArrayFromFileStream(Stream stream, out byte[] buff
             // than the buffer size. The default buffer size is 4KB, so this will incur a 4KB
             // allocation for any files less than 4KB. That's why, for example, the command
             // line compiler actually specifies a very small buffer size.
-            return stream.Read(buffer, 0, length) == length;
+            return stream.TryReadAll(buffer, 0, length) == length;
```

Comment:
Nope. I was not being that clever. TryReadAll seems like a safe replacement to me.

Long version: We want to read the whole file into the provided buffer in one gulp. Specifying a destination buffer of the exact length should be a hint to underlying layers that no additional buffering is required. However, if the implementation chooses to break up the reads for us, then there's not much we can do about it.